### PR TITLE
refactor(ir): split out table count into `CountStar` operation

### PR DESCRIPTION
**Pull request by cpcloud**

### CODE REVIEW

File: ibis/expr/types/relations.py [hunk 1]
```diff
@@ -743,15 +743,20 @@ def filter(
...
         return an.apply_filter(table.op(), predicates).to_expr()
 
-    def count(self) -> ir.IntegerScalar:
+    def count(self, where: ir.BooleanValue | None = None) -> ir.IntegerScalar:
         """Compute the number of rows in the table.
 
+        Parameters
+        ----------
+        where
+            Filter expression
```
Comment (jcrist) [inline ibis/expr/types/relations.py:752]: This seems a bit brief for a new user to understand what `where` is for, could you expand this a bit (or add an example)?

Also ibis docstrings question: for optional parameters, do we usually say "Optional" somewhere in the docstring/say what the default behavior is? Or is that inferred from the function signature?

**Review by jcrist — COMMENTED** [review]
I'm curious, why use this over `table.filter(...).count()`? Just a different way to spell it? Are there other reductions that accept a filter like this?

File: ibis/expr/types/relations.py (hunk 1 above)
Comment (cpcloud) [reply ibis/expr/types/relations.py:752]: > I'm curious, why use this over `table.filter(...).count()`? Just a different way to spell it? Are there other reductions that accept a filter like this?

All of the other reductions accept a filter :)

Comment (cpcloud) [reply ibis/expr/types/relations.py:752]: I can add a more thorough description.

We usually don't say optional unless the behavior is more complex than "has no effect".

**Comment by cpcloud** [comment]
Actually, aren't you using those filters in your datasette blog post?

**Comment by jcrist** [comment]
No, I only use `.filter(...)` for filtering.

**Comment by kszucs** [comment]
@cpcloud looks good to me on principle though a couple of builds are failing

**Comment by kszucs** [comment]
The semantics for filtering reductions are different than applying reductions on a filtered table. As an example see the if aggregation combinator of clickhouse https://clickhouse.com/docs/en/sql-reference/aggregate-functions/combinators/#-if

File: ibis/expr/types/relations.py (hunk 1 above)
Comment (cpcloud) [reply ibis/expr/types/relations.py]: Fixed!

**Pull request merged**